include/odp/api/spec/crypto.h

```diff
@@ -467,7 +483,18 @@ typedef struct odp_crypto_session_param_t {
 
 	/** Authentication algorithm
 	 *
-	 *  Use odp_crypto_capability() for supported algorithms.
+	 *  Select authentication algorithm to be used. ODP_AUTH_ALG_NULL
+	 *  indicates that authentication is disabled. Use
+	 *  odp_crypto_capability() for supported algorithms. Note that some
+	 *  algorithms restrict choice of the pairing cipher algorithm. When
+	 *  single algorithm provides both ciphering and authentication
+	 *  (i.e. Authenticated Encryption), authentication side key
+	 *  (auth_key) and IV (auth_iv) are ignored, and cipher side values are
+	 *  used instead. These algorithms ignore authentication side key
+	 *  and IV: ODP_AUTH_ALG_AES_GCM, ODP_AUTH_ALG_AES_CCM and
+	 *  ODP_AUTH_ALG_CHACHA20_POLY1305. Otherwise, all authentication side
```

Comment:
Do we want/need to enumerate these here? Each implementation chooses which algorithms it implements (as indicated by `odp_crypto_capability()`. It's up to each user of those algorithms to understand their associated usage conventions.

Reply:
These enumerated values change the default usage of API. With all other auth algos, user must fill in all auth parameters in this struct. When auth algo is one of these three, auth_key and auth_iv params are ignored and cipher ones are used instead (as those would be the same anyway).